controller/konnect/konnectextension_controller.go

```diff
@@ -68,7 +86,16 @@ func (r *KonnectExtensionReconciler) SetupWithManager(ctx context.Context, mgr c
 				),
 			),
 		).
-		// TODO: watch secrets https://github.com/Kong/gateway-operator/issues/1210
+		Watches(
+			&corev1.Secret{},
+			handler.EnqueueRequestsFromMapFunc(
+				konnectExtensionReconcileRequestsForSecret(mgr.GetClient()),
+			),
+			builder.WithPredicates(
+				labelSelectorPredicate,
+				predicate.NewPredicateFuncs(secretUsedByKonnectExtension(mgr.GetClient())),
```

Comment:
I believe that this is unnecessary given that we use the same information in the enqueue func.

We're basically checking (and listing) the same thing twice:

- first on the `KonnectExtension` listing using the index that is based on the `.Spec.DataPlaneClientAuth.CertificateSecret.CertificateSecretRef.Name`
- secondly in the `Secret` predicate listing the `KonnectExtension`s using the very same index and checking the `.Spec.DataPlaneClientAuth.CertificateSecret.CertificateSecretRef.Name` field.

I'd suggest to drop this to prevent the unnecessary work. We can add a comment if we'd like to be verbose about this here. (That's where the secret label comes in handy to limit the amount of secrets we'll perform the listing for).

```suggestion
```

Reply:
Done.